src/ai/backend/manager/scheduler/dispatcher.py

```diff
@@ -469,13 +469,33 @@ async def _schedule_single_node_session(
         log_fmt = _log_fmt.get()
         log_args = _log_args.get()
         try:
-            agent_id = scheduler.assign_agent_for_session(candidate_agents, sess_ctx)
-            if agent_id is None:
-                raise InstanceNotAvailable
+            # OP.1415 if configured sess_ctx.agent_id is existed, skip assin_agent_for_session
+            agent_id = None
+            if sess_ctx.agent_id is not None:
+                agent_id = sess_ctx.agent_id
+            else:
+                agent_id = scheduler.assign_agent_for_session(candidate_agents, sess_ctx)
             async with self.db.begin() as agent_db_conn:
-                agent_alloc_ctx = await _reserve_agent(
-                    sched_ctx, agent_db_conn, sgroup_name, agent_id, sess_ctx.requested_slots,
+                query = (
+                    sa.select([agents.c.available_slots])
+                    .select_from(agents)
+                    .where(agents.c.id == agent_id)
                 )
+                available_agent_slots = (await agent_db_conn.execute(query)).scalar()
+                # if pass the available test
+                if available_agent_slots is None:
+                    raise InstanceNotAvailable("There is no available_agent_slots")
+                available_test_pass = False
+                for key in available_agent_slots:
+                    if available_agent_slots[key] >= sess_ctx.requested_slots[key]:
+                        available_test_pass = True
+                        continue
+                    else:
+                        raise InstanceNotAvailable(f"{key} is insufficent.")
```

Comment:
Please include more contextual information.
```suggestion
                        raise InstanceNotAvailable(f"The resource slot {key} in {agent_id} does not have the enough remaining capacity.")

```